File: app/src/main/java/com/example/sendserverdemo/RetrofitService.java
package com.example.sendserverdemo;


import org.jetbrains.annotations.Nullable;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.HeaderMap;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.QueryMap;

public interface RetrofitService {

    @GET("{path}")
    Call<String> get(@HeaderMap Map<String, String> headers, @Path(value = "path", encoded = true) String path, @QueryMap Map<String, String> map);

    @FormUrlEncoded
    @POST("{path}")
    Call<String> post(@Path(value = "path", encoded = true) String path, @FieldMap Map<String, String> map);

}

File: retrofit/src/main/java/com/kcst/retrofit/net/RetrofitService.kt
package com.kcst.retrofit.net

import retrofit2.Call
import retrofit2.http.FieldMap
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.HeaderMap
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.QueryMap

interface RetrofitService {
    @GET("{path}")
    operator fun get(
        @HeaderMap headers: Map<String, String>?,
        @Path(value = "path", encoded = true) path: String,
        @QueryMap map: Map<String, String>
    ): Call<String>

    @FormUrlEncoded
    @POST("{path}")
    fun post(
        @HeaderMap headers: Map<String, String>?,
        @Path(value = "path", encoded = true) path: String,
        @FieldMap map: Map<String, String>
    ): Call<String>
}